crest/Assets/Crest/Crest/Scripts/LodData/RegisterClipSurfaceInput.cs

```diff
@@ -98,9 +235,80 @@ private void LateUpdate()
         protected override bool FeatureEnabled(OceanRenderer ocean) => ocean.CreateClipSurfaceData;
         protected override string RequiredShaderKeywordProperty => LodDataMgrClipSurface.MATERIAL_KEYWORD_PROPERTY;
         protected override string RequiredShaderKeyword => LodDataMgrClipSurface.MATERIAL_KEYWORD;
-
         protected override string MaterialFeatureDisabledError => LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING;
         protected override string MaterialFeatureDisabledFix => LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING_FIX;
+
+        protected override bool RendererRequired => _mode == Mode.Geometry;
+        protected override bool RendererOptional => _mode != Mode.Geometry;
+
+        // Use Unity's UV sphere mesh for gizmos as Gizmos.DrawSphere is too low resolution.
+        static Mesh s_SphereMesh;
+
+        protected new void OnDrawGizmosSelected()
+        {
+            Gizmos.color = GizmoColor;
+
+            if (_mode == Mode.Geometry)
+            {
+                if (TryGetComponent<MeshFilter>(out var mf))
+                {
+                    Gizmos.DrawWireMesh(mf.sharedMesh, 0, transform.position, transform.rotation, transform.lossyScale);
+                }
+
+                return;
+            }
+
+            if (s_Quad == null)
+            {
+                s_Quad = Resources.GetBuiltinResource<Mesh>("Quad.fbx");
+            }
+
+            // Show gizmo for quad which encompasses the shape.
+            Gizmos.matrix = QuadMatrix;
+            Gizmos.DrawWireMesh(s_Quad);
+
+            Gizmos.matrix = transform.localToWorldMatrix;
+
+            switch (_primitive)
+            {
+                case Primitive.Sphere:
+                    if (s_SphereMesh == null)
+                    {
+                        s_SphereMesh = Resources.GetBuiltinResource<Mesh>("New-Sphere.fbx");
+                    }
+
+                    // Render mesh and wire sphere at default size (0.5m radius) which is scaled by gizmo matrix.
+                    Gizmos.DrawMesh(s_SphereMesh, submeshIndex: 0, Vector3.zero, Quaternion.identity, Vector3.one);
+                    Gizmos.DrawWireSphere(Vector3.zero, 0.5f);
+                    break;
+                case Primitive.Cube:
+                    // Render mesh and wire box at default size which is scaled by gizmo matrix.
+                    Gizmos.DrawCube(Vector3.zero, Vector3.one);
+                    Gizmos.DrawWireCube(Vector3.zero, Vector3.one);
+                    break;
+                default:
+                    Debug.LogError("Crest: Not a valid primitive type!");
+                    break;
+            }
+        }
 #endif
     }
+
+    public partial class RegisterClipSurfaceInput : ISerializationCallbackReceiver
+    {
+        public void OnAfterDeserialize()
+        {
+            if (_version == 0)
```

Comment:
this looks great!

contemplating if this should dirty the component. i think dirtying would push the version forward. on the flipside, in production scenarios, an artist finding the scene is dirtied for unknown reasons causes confusion. i think maybe lets leave it the way you have it.

perhaps its worth adding a comment above this if() statement with the date and reason (default mode changed from geo to primitive). seems useful to have this logged in an easily readable way, and makes decisions about removing migration code in the future easy.

Reply:
> artist finding the scene is dirtied for unknown reasons causes confusion

I dislike this too. Also, calling dirty cannot be done in these methods so I would need to call it in Start.

> perhaps its worth adding a comment

Makes sense. Done.